gcp_variant_transforms/libs/generate_report.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generates conflicts report."""
+
+import csv
+import os
+from typing import Dict, List, Union  # pylint: disable=unused-import
+
+from gcp_variant_transforms.beam_io.vcf_header_io import VcfHeader  # pylint: disable=unused-import
+from gcp_variant_transforms.beam_io.vcf_header_io import VcfParserHeaderKeyConstants
+from gcp_variant_transforms.transforms.merge_header_definitions import Definition  # pylint: disable=unused-import
+from gcp_variant_transforms.transforms.merge_header_definitions import VcfHeaderDefinitions  # pylint: disable=unused-import
+
+
+class ConflictsReporter(object):
+  """Generates the header conflicts report."""
+
+  _HEADER_LINE = ['ID', 'Conflicts', 'Proposed Solution']
+  _REPORT_NAME = 'header_conflicts_report.csv'
+  _NO_CONFLICTS_MESSAGE = ['No conflicts found.']
+  _NO_SOLUTION_MESSAGE = 'Not resolved'
+
+  def __init__(self, directory=''):
```

Comment:
If we decide to remove `directory` from this class, then I think this can just be a module instead of a class (i.e. you'll have one public `generate_conflicts_report(header_defs, resolved_header, path)` method)